llama-index-server/llama_index/server/gen_ui/main.py

```
@@ -0,0 +1,419 @@
+
+def pre_run_checks() -> None:
+    if not os.getenv("ANTHROPIC_API_KEY"):
+        raise ValueError(
+            "Anthropic API key is not set. Please set the ANTHROPIC_API_KEY environment variable."
+        )
+    try:
+        from llama_index.llms.anthropic import Anthropic  # noqa: F401
+    except ImportError:
+        raise ValueError(
+            "Anthropic package is not installed. Please install it with `poetry add llama-index-llms-anthropic` or `pip install llama-index-llms-anthropic`."
+        )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)